### PR TITLE
Set up sub-apps for OEB 252 and CS 1090a

### PR DESCRIPTION
**Pull request by jaguillette**
This PR uses the sub-app structure to create two different Conda-based Jupyter apps, one for CS 109a, and the other for OEB 252. They each have their own conda and spack definitions, which are configured in the prod environment already. I've tested this in prod as a sandbox app, which is how I landed on the pip installation of dependencies, since conda itself wasn't getting a functioning numpy somehow.

This would replace the branched version of the app currently in place in prod, since we can make it work via a sub-app instead. Note that this app configuration also hard-codes the course IDs that should have access to the app into the form.yml files for the app configuration. I've commented them with the names of the courses, but this still isn't a centralized place to put them.

As a follow up when this PR is approved, there should be an update to the IAC repo to set up this app as part of the system app setup.

This setup is similar to the approach described in [the OOD Jupyter Lab spack implementation PR 3](https://github.com/Harvard-ATG/ood-jupyterlab-spack/pull/3).

### CODE REVIEW

**Review by Tanvez — APPROVED** [review]
Looks good to me!

File: local/cs1090a.yml.erb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  # @example Do not load any modules
+  #     modules: ""
+  # @example Using default python module
+  #     modules: "python"
+  # @example Using specific python module
+  #     modules: "python/3.5"
+  # @example Using combination of modules
+  #     modules: "python/3.5 cuda/8.0.44"
+
+  environment: "cs109a"
```
Comment (arthurian) [inline local/cs1090a.yml.erb:53]: In the [form.yml](https://github.com/Harvard-ATG/ood-jupyterlab-spack-conda/blob/course/cs109a/form.yml#L19-L27) for `course/cs109a`, we have this toggling between `course|global` and then the script uses that to set the appropriate `spack` and `conda` values in the script. I think for this to work, we'll need to adapt this slightly.

Comment (jaguillette) [reply local/cs1090a.yml.erb:53]: Oh, yep you're totally right. I need to bring in the changes that you had to the other branch. Working on bringing those in, will ping when they're incorporated.

**Comment by jaguillette** [comment]
Okay, I've set things up to mirror the cs109a fork, with a global and course flag available to the sub-apps, and environment names that get passed down from the form.yml so the different sub-apps can use them. I have this branch set up in stage, and everything works except the CS 109a course folder environment, which I was struggling to remember how to get to re-install conda so it wouldn't use the global conda base. I know I saw a writeup of how @arthurian did that somewhere, but I can't seem to find it right now. I'll keep at it tomorrow, and perhaps we can take a look during kibitz.

**Comment by jaguillette** [comment]
Update this morning: I was able to get the CS 109a environment to have its own miniconda, not reusing the global one, by uninstalling miniconda from the environment with `spack remove miniconda3`, then re-adding it with `spack add miniconda3` and installing it again, this time specifying the `--fresh-roots` option: `spack install --fresh-roots miniconda3`. Note that putting the `--fresh-roots` option after `miniconda3` treats the option as a flag for the software, rather than an install command option.

**Review by arthurian — APPROVED** [review]
Looks good @jaguillette! :+1:

**Pull request merged**